pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -353,3 +515,87 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 	log.Debugf("cluster flavor %q not recognised. Defaulting to false", c.clusterFlavor)
 	return false
 }
+
+// IsFakeAttachAllowed checks if the volume is eligible to be fake attached and returns a bool value
+func (c *K8sOrchestrator) IsFakeAttachAllowed(ctx context.Context, volumeID string, volumeManager cnsvolume.Manager) (bool, error) {
+	log := logger.GetLogger(ctx)
+	//Check pvc annotations
+	pvcAnn, err := c.getPVCAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get pvc annotations")
+		return false, err
+	}
+
+	if val, found := pvcAnn[common.AnnIgnoreInaccessiblePV]; found && val == "yes" {
+		log.Debugf("Found %s annotation on pvc for volume %s set to yes", common.AnnIgnoreInaccessiblePV, volumeID)
```

Comment:
This log message may not be needed since Line: 532 also gets printed

Reply:
Or we can combine both.
`log.Debugf("Found %s annotation on PVC set to yes for CNS volume %s. Checking volume health on the volume", common.AnnIgnoreInaccessiblePV, volumeID)`